src/pages/workspaces/workspace/jobHistory/WorkflowDashboard.js

```diff
@@ -0,0 +1,181 @@
+import _ from 'lodash/fp'
+import { Fragment, useRef, useState } from 'react'
+import { div, h, span } from 'react-hyperscript-helpers'
+import ReactJson from 'react-json-view'
+import * as breadcrumbs from 'src/components/breadcrumbs'
+import Collapse from 'src/components/Collapse'
+import { ClipboardButton, Link } from 'src/components/common'
+import { centeredSpinner, icon } from 'src/components/icons'
+import {
+  collapseCromwellExecutionStatus, failedIcon, makeSection, makeStatusLine, runningIcon, statusIcon,
+  submittedIcon, successIcon, unknownIcon, workflowDetailsBreadcrumbSubtitle
+} from 'src/components/job-common'
+import UriViewer from 'src/components/UriViewer'
+import WDLViewer from 'src/components/WDLViewer'
+import { Ajax } from 'src/libs/ajax'
+import { bucketBrowserUrl } from 'src/libs/auth'
+import { getConfig } from 'src/libs/config'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+import { wrapWorkspace } from 'src/pages/workspaces/workspace/WorkspaceContainer'
+
+
+const styles = {
+  sectionTableLabel: { fontWeight: 600 }
+}
+
+// Note: this can take a while with large data inputs. Consider memoization if the page ever needs re-rendering.
+const groupCallStatuses = _.flow(
+  _.values,
+  _.flattenDepth(1),
+  _.countBy(a => collapseCromwellExecutionStatus(a.executionStatus))
+)
+
+const statusCell = ({ calls }) => {
+  const { succeeded, failed, running, submitted, ...others } = groupCallStatuses(calls)
+
+  const makeRow = (count, icon, text) => {
+    return !!count && div({ style: { display: 'flex', alignItems: 'center', marginTop: '0.25rem' } }, [
+      icon,
+      span([` ${count} ${text}`])
+    ])
+  }
+
+  return h(Fragment, [
+    makeRow(submitted, submittedIcon(), 'Submitted'),
+    makeRow(running, runningIcon(), 'Running'),
+    makeRow(succeeded, successIcon(), 'Succeeded'),
+    makeRow(failed, failedIcon(), 'Failed'),
+    _.map(([name, count]) => makeRow(count, unknownIcon(), name), _.toPairs(others))
+  ])
+}
+
+const WorkflowDashboard = _.flow(
+  Utils.forwardRefWithName('WorkflowDashboard'),
+  wrapWorkspace({
+    breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
+    title: 'Job History', activeTab: 'job history'
+  })
+)((props, ref) => {
+  const { namespace, name, submissionId, workflowId, workspace: { workspace: { bucketName } } } = props
+
+  /*
+   * State setup
+   */
+  const [workflow, setWorkflow] = useState({})
```

Comment:
To allow us to use the standard pattern of undefined as not yet loaded:
```suggestion
  const [workflow, setWorkflow] = useState()
```

Reply:
Doing this caused problems later on due to destructuring on the `undefined`:
```
const { end, failures, start, status, workflowLog, workflowName, submittedFiles: { workflow: wdl } = {} } = workflow
```

I could get around that with a ternary guard on the destructuring, but this now feels a little... obtuse? Interested in your thoughts:
```
const { end, failures, start, status, workflowLog, workflowName, submittedFiles: { workflow: wdl } = {} } = workflow ? workflow : {}
```

Reply:
This was answered in your next comment 🤦‍♂️